extensions/azure/tests/ListAzureDataLakeStorageTests.cpp

```diff
@@ -43,7 +43,7 @@ class ListAzureDataLakeStorageTestsFixture {
     auto mock_data_lake_storage_client = std::make_unique<MockDataLakeStorageClient>();
     mock_data_lake_storage_client_ptr_ = mock_data_lake_storage_client.get();
     auto uuid = utils::IdGenerator::getIdGenerator()->generate();
-    auto impl = std::unique_ptr<minifi::azure::processors::ListAzureDataLakeStorage>(
+    auto impl = std::unique_ptr<minifi::azure::processors::ListAzureDataLakeStorage>(  // NOLINT(clang-analyzer-cplusplus.NewDeleteLeaks)
```

Comment:
does make_unique not work here?

Reply:
Unfortunately no due to the constructor overload being private and only the friend class `ListAzureDataLakeStorageTestsFixture` can instantiate it, the `unique_ptr` cannot.